src/share/mkjail/create.sh

```diff
@@ -80,10 +80,9 @@ if [ x"${fflag}" = x1 ] && [ ! -d /var/db/mkjail/flavours/${FLAVOUR} ]; then
 fi
 
 # Create the ZFS filesystem
-echo "Creating ${ZPOOL}/jails/${JAILNAME}..."
-zfs create -p -o mountpoint=/jails ${ZPOOL}/jails
-zfs create -p ${ZPOOL}/jails/${JAILNAME}
-zfs set mkjail:version=${VERSION} ${ZPOOL}/jails/${JAILNAME}
+echo "Creating ${ZPOOL}${JAILDATASET}/${JAILNAME}..."
+zfs create "${ZPOOL}${JAILDATASET}/${JAILNAME}"
```

Comment:
Why drop the `-p` and `-o` options?

Reply:
I've moved the first create command into the `_init()` function, see my comment below.

For the second create command I removed the `-p` option because if the parent dataset doesn't exist it means the first create command failed and the script should be exiting.